tests/common/helpers/snmp_helpers.py

```diff
@@ -10,20 +10,34 @@
 
 DEF_WAIT_TIMEOUT = 300
 DEF_CHECK_INTERVAL = 10
+SNMP_SUBAGENT_WAIT_TIMEOUT = 120
+SNMP_SUBAGENT_CHECK_INTERVAL = 5
 
 global_snmp_facts = {}
 
 
+def is_snmp_subagent_running(duthost):
+    cmd = "docker exec snmp supervisorctl status snmp-subagent"
+    output = duthost.shell(cmd)
+    if "RUNNING" in output["stdout"]:
+        return True
+    return False
+
+
 def _get_snmp_facts(localhost, host, version, community, is_dell, include_swap, module_ignore_errors):
     snmp_facts = localhost.snmp_facts(host=host, version=version, community=community, is_dell=is_dell,
                                       module_ignore_errors=module_ignore_errors, include_swap=include_swap)
     return snmp_facts
 
 
-def _update_snmp_facts(localhost, host, version, community, is_dell, include_swap):
+def _update_snmp_facts(localhost, host, version, community, is_dell, include_swap, duthost):
     global global_snmp_facts
 
     try:
+        pytest_assert(
```

Comment:
Just want to double confirm since I'm not sure about the realationship between SNMP Sub-Agent and SNMP_FACTS.

Is SNMP Sub-Agent ready(is_snmp_subagent_running) a prerequisites for getting snmp facts(_get_snmp_facts)?
If so, doesn't that mean when we can get snmp_facts by _get_snmp_facts(), the snmp sub-agent is ready already?

Reply:
`_get_snmp_facts` is using `snmp_facts` which is an Ansible module & this doesn't have a pre-requisite for `snmp-subagent` to be ready; if there's a delay in `snmp-subagent` to be up & by then `snmp_facts` is already called then either we see "Timeout issues" or "Missing Information" in `snmp_facts` output.

Reply:
Then if we can get expected output from `snmp_facts` means snmp-subagent is ready already?
If the readiness of '_get_snmp_facts' implicitly include the readiness of snmp-subagent,
We already have the logic of wait_for `_update_snmp_facts` ?

Or do you mean even if the "Timeout issues" or "Missing Information" in `snmp_facts` output, `except RunAnsibleModuleFail as e:` can't catch them?